pkg/virt-controller/watch/application.go

```diff
@@ -0,0 +1,129 @@
+package watch
+
+import (
+	"flag"
+	golog "log"
+	"net/http"
+	"strconv"
+
+	"github.com/emicklei/go-restful"
+	clientrest "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+
+	"k8s.io/client-go/util/workqueue"
+
+	"k8s.io/client-go/kubernetes"
+
+	kubeinformers "kubevirt.io/kubevirt/pkg/informers"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/logging"
+	"kubevirt.io/kubevirt/pkg/virt-controller/rest"
+	"kubevirt.io/kubevirt/pkg/virt-controller/services"
+)
+
+var (
+	flags           Flags = Flags{}
```

Comment:
These 17 variables should not be globals. This is generally an anti-pattern in go and can lead to potential race conditions.

What we should do is create a type that contains all of these and pass (a pointer to) an instance of that as a parameter to Execute() or other functions that need it.